feat(ConversationIcon): highlight public conversations

Antreesy opened the pull request:
### ☑️ Resolves

* Conversation (NcListItem) now indicates, if conversation is public (available for guests by link)
* Place is details above the unread counter

* To discuss:
  * Icon for federated conversations:
    * Same place?
    * Should overwrite public icon?
    * Which icon to use? https://pictogrammers.com/library/mdi/icon/shield-link-variant/ as option

## 🖌️ UI Checklist

### 🖼️ Screenshots / Screencasts

| :sun_with_face: Light theme | :new_moon: Dark theme |
|------------|----------|
| Default with `--color-primary-element-text` for active            | `--color-text-maxcontrast` for others         |
![image](https://github.com/nextcloud/spreed/assets/93392545/cc335ab3-f518-48d5-bd0e-2934b8f7d1b8) | ![image](https://github.com/nextcloud/spreed/assets/93392545/17c307d8-b049-4813-9118-29f5f9fb97a3)

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [x] 🖌️ Design was reviewed, approved or inspired by the design team

marcoambrosini reviewed (commented):
I think we should avoid adding this information on the avatar @Antreesy 
Let me think about this and will come up with an alternative proposal soon.

Antreesy commented:
Proposal from Marco:
![image](https://github.com/nextcloud/spreed/assets/93392545/fc69eede-b377-46b6-967b-b445d9e233e8)

Requires an upstream modification of `NcListItem` component

Antreesy commented:
Rebased and updated screenshots.
Don't think that group conversations require that 'multiple-account' icon, to not overwhelm the sidebar, but we could discuss that

DorraJaouad reviewed (approved):
Code-wise, it's fine ✅ 
We can keep the same icon for federated conversations or it should be changed for some reason ?

marcoambrosini commented:
I think it would be nice info @Antreesy , shouldn't overwhelm much

marcoambrosini commented:
@Antreesy could we use the same color as the title for those icons?
